app/assets/javascripts/discourse/tests/unit/ember/resolver-test.js

```diff
@@ -202,63 +271,248 @@ module("Unit | Ember | resolver", function (hooks) {
       "admin/templates/foo",
       "adminBar",
       "admin_bar",
       "admin.bar",
       "admin/templates/bar",
+      "admin/templates/dashboard_general",
+      "admin-baz-qux",
+      "javascripts/admin/plugin-template",
     ]);
 
+    // Switches prefix to admin/templates when camelized
     lookupTemplate(
       assert,
       "template:adminFoo",
       "admin/templates/foo",
       "when prefix is separated by camel case"
     );
+
+    // Switches prefix to admin/templates when underscored
     lookupTemplate(
       assert,
       "template:admin_foo",
       "admin/templates/foo",
       "when prefix is separated by underscore"
     );
+
+    // Switches prefix to admin/templates when dotted
     lookupTemplate(
       assert,
       "template:admin.foo",
       "admin/templates/foo",
       "when prefix is separated by dot"
     );
 
+    // Doesn't match unseparated prefix
     lookupTemplate(
       assert,
       "template:adminfoo",
       undefined,
       "but not when prefix is not separated in any way"
     );
+
+    // Prioritized the default match when camelized
     lookupTemplate(
       assert,
       "template:adminBar",
       "adminBar",
       "but not when template with the exact camel cased name exists"
     );
+
+    // Prioritized the default match when underscored
     lookupTemplate(
       assert,
       "template:admin_bar",
       "admin_bar",
       "but not when template with the exact underscored name exists"
     );
+
+    // Prioritized the default match when dotted
     lookupTemplate(
       assert,
       "template:admin.bar",
       "admin.bar",
       "but not when template with the exact dotted name exists"
     );
+
+    lookupTemplate(
+      assert,
+      "template:admin-dashboard-general",
+      "admin/templates/dashboard_general",
+      "finds namespaced and underscored version"
+    );
+
+    lookupTemplate(
+      assert,
+      "template:admin-baz/qux",
+      "admin-baz-qux",
+      "also tries dasherized"
+    );
+
+    lookupTemplate(
+      assert,
+      "template:admin-plugin/template",
+      "javascripts/admin/plugin-template",
+      "looks up templates in pluginss"
+    );
+  });
+
+  test("resolves component templates with 'admin' prefix to 'admin/templates/' namespace", function (assert) {
+    setTemplates([
+      "admin/templates/components/foo",
+      "components/bar",
+      "admin/templates/components/bar",
+    ]);
+
+    // Looks for components in admin/templates
+    lookupTemplate(
+      assert,
+      "template:components/foo",
+      "admin/templates/components/foo",
+      "uses admin template component when no standard match"
+    );
+
+    // Prioritized non-admin component
+    lookupTemplate(
+      assert,
+      "template:components/bar",
+      "components/bar",
+      "uses standard match when both exist"
+    );
+  });
+
+  // TODO: Is this necessary? IIRC internal Ember code will use the top level loading template by default
```

Comment:
Are you asking whether the test is necessary? Or whether some of our custom resolver logic is necessary?

Either way, it would be good to work this out so that we can avoid merging a TODO comment.

Reply:
I'm asking if this logic is necessary. I'll investigate more.

Reply:
We should probably investigate this further. I suggest doing it as a follow-up to this PR as we work to standardize things further.